apps/prairielearn/src/components/QuestionsTable.html.ts

```diff
@@ -0,0 +1,185 @@
+import { html, HtmlSafeString } from '@prairielearn/html';
+import { nodeModulesAssetPath, compiledStylesheetPath } from '../lib/assets';
+import { EncodedData } from '@prairielearn/browser-utils';
+import { CourseInstance } from '../lib/db-types';
+import { QuestionsTableData } from '../models/questions';
+
+export function QuestionsTable({
+  questions,
+  showAddQuestionButton,
+  current_course_instance,
+  course_instances,
+  urlPrefix,
+  plainUrlPrefix,
+  __csrf_token,
+}: {
+  questions: QuestionsTableData[];
+  showAddQuestionButton: boolean;
+  current_course_instance: CourseInstance;
+  course_instances: CourseInstance[];
+  urlPrefix: string;
+  plainUrlPrefix: string;
+  __csrf_token: string;
+}): HtmlSafeString {
+  const has_legacy_questions = questions.some((row) => row.display_type !== 'v3');
+  const course_instance_ids = (course_instances || []).map((course_instance) => course_instance.id);
+  // Importing javascript using <script> tags as below is *not* the preferred method, it is better to directly use 'import'
+  // from a javascript file. However, bootstrap-table is doing some hacky stuff that prevents us from importing it this
+  return html`
+    <script src="${nodeModulesAssetPath('bootstrap-table/dist/bootstrap-table.min.js')}"></script>
+    <script src="${nodeModulesAssetPath(
+        'bootstrap-table/dist/extensions/sticky-header/bootstrap-table-sticky-header.min.js',
+      )}"></script>
+    <script src="${nodeModulesAssetPath(
+        'bootstrap-table/dist/extensions/filter-control/bootstrap-table-filter-control.min.js',
+      )}"></script>
+
+    ${EncodedData(
+      { course_instance_ids, showAddQuestionButton, urlPrefix, plainUrlPrefix },
+      'questions-table-data',
+    )}
+
+    <link href="${compiledStylesheetPath('questionsTable.css')}" rel="stylesheet" />
```

Comment:
Since these usually go in the head, would it make sense to add them to a separate function? I mean the scripts and link, not encoded data.

Reply:
I have put the `scripts` into a seperate function. I like that better actually because its more clear what the warning comment is referring to. 

For the stylesheet link, do you want me to put that in the head? I can put it in the head in `instructorQuestions.html.ts` if you want. It seems to make sense to put it here to me since it is part of the component. 

I just have no idea what the convention for this is, I can't find any examples of this whatsoever in PrairieLearn.

Reply:
By a separate function I meant something like separating `export function QuestionsTable` and `export function QuestionsTableHead`, and call both from whatever uses the component (one in the head, one in the body). Does that make sense?

Reply:
done, thanks